app/queries/latest_metrics.rb

```diff
@@ -0,0 +1,16 @@
+class LatestMetrics < SimpleDelegator
+  def find
+    Metric.find_by_sql([sql, id])
+  end
+
+  def sql
+    'SELECT m1.*
+    FROM metrics m1
+    WHERE m1.branch_id = ?
```

Comment:
I think its okay, but maybe making a group by and ordering by created_at each of them could be even more clear and more efficient